.github/actions/e2e-download-uberjar/action.yml

```diff
@@ -0,0 +1,75 @@
+name: Download uberjar for E2E tests
+description: Download either a freshly built artifact or the one built in one of the previous commits.
+inputs:
+  edition:
+    description: Metabase edition.
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - name: Download previously stored uberjar
+      uses: actions/github-script@v6
+      with:
+        script: |
+          const fs = require('fs');
+          const execSync = require("child_process").execSync;
+
+          const getCommit = (ref = "HEAD") => {
+            const [commit] = execSync(`git rev-parse ${ref}`, { encoding: "utf8" }).split(
+              "\n",
+            );
+            return commit;
+          };
+
+          const baseConfig = {
+            owner: context.repo.owner,
+            repo: context.repo.repo,
+          };
+
+          async function getArtifact(commit, depth = 0) {
+            if (depth > 10) {
+              throw new Error("Couldn't find the artifact!");
+            }
```

Comment:
i'm not sure if 10 is deep enough, especially since we're fetching 20. I can easily imagine pushing 10 different tes-only commits to make tests stop flaking.

Reply:
Increased to 20.